### DISCUSSION TITLE
New Video Panel [Different UX for better usability]

### DISCUSSION THREAD

**Discussion by bachig26** [FreeTubeApp/FreeTube#2016] [General]
**Description:**

With this new panel, once a video is clicked to view, the **_panel pops up as a new column similar to a File Explorer Column view and lets the video playable and subsequently view the video description and comments_**.

By using this feature, we can explore new searches and channels while watching the content, **_also it needs a new toggle icon embedded near the video player pip icon or so to switch between the original video view and this new view _**.

The new feature is to implement a _**toggle function**_ in Settings to _**Turn this new video panel on or off**_ as user interest.

**Screenshots:**
_**Current Layout, which invokes the video player page [similar to YouTube]**_
![FreeTube_t2FV3hBzhq](https://user-images.githubusercontent.com/96393348/150274413-0581b78d-9299-49ca-9982-a5e6234925a3.png)

**_New Layout, opens a new panel adjacent to the above content in a manner of a column view_**
![New Video Panel](https://user-images.githubusercontent.com/96393348/150276929-078043b9-4f2c-46b2-bddd-6a57e0ec8485.jpg)

**Comment by bachig26** [comment]
Hey Guys, I would like to discuss this idea implementation to the project if possible.
The **screenshot stated as "New Layout" is edited with paint application**, as I thought how the UI will look like.

I would like to hear your opinions regarding this one, so I didn't create this as an issue because I didn't know if this one will go for the PR not.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
First of all please dont ping everyone like. I would understand of u do that if it was something urgent like security issus or something that leaks private information but not this. Everybody will eventually look at it even when u dont ping them.

**Reply by bachig26** [reply]
sorry

**Comment by PikachuEXE** [comment]
What value does this new layout provide?
If the user has a big screen then opening videos in new windows is already an option (with playlist & recommended videos)

**Reply by bachig26** [reply]
Regarding the new windows, I don't know about the behind-the-screen code structure of how it runs. I believe it creates and runs as a second instance.

In case of value to this new layout, I don't have exact thought to give more convincing facts but based on my experience I like concise the work environment within a single-window as possible, maybe have some tabs but no additional windows. also, it will be running as a single instance and as a multi-tasking functionality or something like that.
However, this will give more flexibility or versatility in the whole search and watch space.

Note: Also, I know that the PiP now has some bug, that lets users watch a video and can go to a different profile or channel or different video, and still plays both contents in a single-window simultaneously.

**_Also, I believe there needs to be a customization factor that can be different, useful, efficient that comes handy in certain cases. for Example, I use Vivaldi as my browser and I am kind liked to have the customization thing in my work environment._**

**Reply by PikachuEXE** [reply]
New windows in this app is like new browser windows (as it's based on electron which is based on chromium).

I agree that the addition of tabs can be a great UX improvement and the discussion can be found on https://github.com/FreeTubeApp/FreeTube/issues/333
I think I would not consider this proposed feature until 1.0 or after tabs implemented

Right now the new feature dev focus would be on playlist (not the current one)
We are a small team so we can only focus on limited things (plus bug fixes and small feature additions)

Let's see if other devs want to express their thoughts

**Reply by bachig26** [reply]
I can understand that.

Maybe, if you think this feature can be used in the future, then you can mark it where it can be accessible or something like that.

Also, if possible I would like to know what kind of major feature or change are you guys working on for the next beta version?

**Reply by PikachuEXE** [reply]
https://blog.freetubeapp.io/behind-the-scenes-improving-project-structure-and-other-updates

> One of the major features that everyone is looking forward to has been the full implementation of playlists.

**Reply by bachig26** [reply]
will there be a complete refactoring happen when pushing to v1.0 or before?

**Reply by PikachuEXE** [reply]
I have no idea~

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
My thoughts are basically the same as @PikachuEXE

**Reply by bachig26** [reply]
I just thought that if tab functionality is enabled, will there be a possibility of tab tiling also added?

**Reply by PikachuEXE** [reply]
Possible? I guess
Difficult? I guess too
Anyway I have no idea what technical limitations there are and they might be removed/eased later
I can't give any accurate/confident guess right now

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
I'm going ahead and close this because there are no plans to implement this. If anyone wants to implement this please feel free to leave a comment or to create a PR.
